### PR TITLE
🎓 Sphere class

### PR DESCRIPTION
**Pull request by jameshughes89**
### What

Sphere class that makes use of a `Point3D`

### Testing

:+1:
But you will notice a lack of `assert` tests. This has the benefit of
- This topic leads into the unittest topic
- I don't have to write them

### CODE REVIEW

File: src/sphere.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import math
+
+
+class Sphere:
+    """
+    Class for managing Spheres within a 3D space.
+    """
+
+    def __init__(self, centre_point, radius):
```
Comment (twentylemon) [inline src/sphere.py:9]: Why no type hints?

**Review by twentylemon — APPROVED** [review]

File: src/sphere.py (hunk 1 above)
Comment (jameshughes89) [reply src/sphere.py]: I added the ones I could, buy adding more requires `typing`. That said, if I am introducing `unittest`, I could do `typing` with it like I did in the original testing and type hints topic early in the course.

Comment (jameshughes89) [reply src/sphere.py]: `float` was added, but to use `Point3D`, I need typing

```python
from typing import Type
...
    def __init__(self, centre_point: Type[Point3D], radius: float):
```

Comment (twentylemon) [reply src/sphere.py]: [nope](https://github.com/duck-dynasty/duckbot/blob/main/duckbot/slash/slash_command_handler.py#L17)

Comment (jameshughes89) [reply src/sphere.py]: my life has been a lie

Comment (twentylemon) [reply src/sphere.py]: [Type[T]](https://docs.python.org/3/library/typing.html#typing.Type) is for generic types. Classes, you can just use use the class name.

Comment (jameshughes89) [reply src/sphere.py]: Gonna' go back and add type hints to previous classes now.

File: src/sphere.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    def diameter(self) -> float:
+        return 2 * self.radius
+
+    def surface_area(self) -> float:
+        return 4 * math.pi * self.radius**2
+
+    def volume(self) -> float:
+        return (4 / 3) * math.pi * self.radius**3
+
+    def distance_between_centres(self, other: Point3D) -> float:
```
Comment (twentylemon) [inline src/sphere.py:26]: `other` is presumably a sphere, not a point

**Pull request merged**